### PR TITLE
Yndrev 12

### PR DESCRIPTION
**Pull request by Olusheenor**
initial draft for sample table geolocation before I implement others

### CODE REVIEW

**Review by Zifah — APPROVED** [review]
Good work. Looks good to me.

File: Api/Controllers/MigratorController.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        private readonly SQLToMongoMigrator _sQLToMongoMigrator;
+        public MigratorController(SQLToMongoMigrator sQLToMongoMigrator)
+        {
+            _sQLToMongoMigrator = sQLToMongoMigrator;
+        }
+        [HttpGet("MigrateGeoLocation")]
+        public IActionResult MigrateGeoLocation()
+        {
+           StringBuilder results = new StringBuilder();
+           results.AppendLine(_sQLToMongoMigrator.MigrateGeolocation());
```
Comment (Zifah) [inline Api/Controllers/MigratorController.cs:20]: Why did you decide to use StringBuilder here? As far as I know, the benefits of using it only materialize in a situation where you're concatenating multiple strings.

**Review by Zifah — APPROVED** [review]

File: Api/Controllers/MigratorController.cs (hunk 1 above)
Comment (Olusheenor) [reply Api/Controllers/MigratorController.cs:20]: I plan to use one end point for all of them so will build more strings eventually.

File: Application/Migrator/SQLToMongoMigrator.cs [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+            List<GeoLocation> documentsToInsert = geolocation.Select(s=> new GeoLocation { Place =s.place, Region =s.region }).ToList();
+
+            _geolocationCollection.DeleteMany(FilterDefinition<GeoLocation>.Empty);
+            _geolocationCollection.InsertMany(documentsToInsert);
+            return $"{documentsToInsert.Count} geolocation records inserted into MongoDB successfully.";
+            
+        }
+        
+
+        //TODO handle more 1 to many mappings i.e  duplicates, feedback, embedded video
```
Comment (Zifah) [inline Application/Migrator/SQLToMongoMigrator.cs:53]: You can ignore duplicates; no use importing them from the old database.

File: Application/Migrator/SQLToMongoMigrator.cs [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+        {
+            switch (input)
+            {
+                case "PUBLISHED": return State.PUBLISHED;
+                case "NEW": return State.NEW;
+                case "UNPUBLISHED": return State.UNPUBLISHED;
+                case "MODIFIED": return State.MODIFIED;
+
+                default:
+                    return State.PUBLISHED;
```
Comment (Zifah) [inline Application/Migrator/SQLToMongoMigrator.cs:122]: Names without a status should have status NEW. Other than that, looks good to me. Feel free to merge in after making that change.

File: Application/Migrator/SQLToMongoMigrator.cs [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+                item.etymology = etymology.Where(f => f.name_entry_id == item.id).ToList()
+                    .Select(s => new Etymology(s.part, s.meaning) { }).ToList();
+
+                item.geoLocations = geolocationPlace.Where(d=>d.name_entry_id == item.id).
+                    Select(a=> new GeoLocation() { Place =a.place, Region = a.region}).ToList();
+
+                item.feedbacks = feedbacks.Where(d => d.name == item.name).
+                    Select(a => new Feedback() { Content = a.feedback }).ToList();
+                            
+                item.embeddedVideo = videos.Where(d=>d.name_entry_id == item.id).
```
Comment (Olusheenor) [inline Application/Migrator/SQLToMongoMigrator.cs:91]: not sure of what the VideoId maps to just made it url..

**Review by Zifah — APPROVED** [review]
Thank you. All looks good to me. And yes, URL maps to VideoId for Videos.

Does this cover all the tables?

**Review by Zifah — DISMISSED** [review]

**Review by Zifah — APPROVED** [review]

**Review by Zifah — APPROVED** [review]

**Pull request merged**